src/internal/observable/fromEvent.ts

```diff
@@ -200,43 +203,91 @@ export function fromEvent<T>(
     return fromEvent<T>(target, eventName, options as EventListenerOptions | undefined).pipe(mapOneOrManyArgs(resultSelector));
   }
 
-  return new Observable<T>((subscriber) => {
-    const handler = (...args: any[]) => subscriber.next(args.length > 1 ? args : args[0]);
-
-    if (isEventTarget(target)) {
-      target.addEventListener(eventName, handler, options as EventListenerOptions);
-      return () => target.removeEventListener(eventName, handler, options as EventListenerOptions);
-    }
-
-    if (isJQueryStyleEventEmitter(target)) {
-      target.on(eventName, handler);
-      return () => target.off(eventName, handler);
-    }
-
-    if (isNodeStyleEventEmitter(target)) {
-      target.addListener(eventName, handler);
-      return () => target.removeListener(eventName, handler);
-    }
+  // Figure out our add and remove methods. In order to do this,
+  // we are going to analyze the target in a preferred order, if
+  // the target matches a given signature, we take the two "add" and "remove"
+  // method names and apply them to a map to create opposite versions of the
+  // same function. This is because they all operate in duplicate pairs,
+  // `addListener(name, handler)`, `removeListener(name, handler)`, for example.
+  // The call only differs by method name, as to whether or not you're adding or removing.
+  const [add, remove] =
+    // If it is an EventTarget, we need to use a slightly different method than the other two patterns.
+    isEventTarget(target)
+      ? eventTargetMethods.map((methodName) => (handler: any) => target[methodName](eventName, handler, options as EventListenerOptions))
+      : // In all other cases, the call pattern is identical with the exception of the method names.
+      isNodeStyleEventEmitter(target)
+      ? nodeEventEmitterMethods.map(toCommonHandlerRegistry(target, eventName))
+      : isJQueryStyleEventEmitter(target)
+      ? jqueryMethods.map(toCommonHandlerRegistry(target, eventName))
+      : [];
 
+  // If add is falsy, it's because we didn't match a pattern above.
+  // Check to see if it is an ArrayLike, because if it is, we want to
+  // try to apply fromEvent to all of it's items. We do this check last,
+  // because there are may be some types that are both ArrayLike *and* implement
+  // event registry points, and we'd rather delegate to that when possible.
+  if (!add) {
     if (isArrayLike(target)) {
-      return (mergeMap((subTarget: any) => fromEvent(subTarget, eventName, options as any))(internalFromArray(target)) as Observable<
-        T
-      >).subscribe(subscriber);
+      return mergeMap((target: any) => fromEvent(target, eventName, options as EventListenerOptions))(
```

Comment:
issue: `target` is effecting a lint failure, as it shadows a variable in an outer scope. FWIW, I have little patience for that rule and don't use it.